_15thnight/core.py

```diff
@@ -24,10 +25,10 @@ def send_out_alert(alert_form):
     providers = User.providers_with_services(need_ids)
     for provider in providers:
         needs_provided = [
-            need for need in provider.services if need.id in need_ids
+            need_ for need_ in provider.services if need_.id in need_ids
```

Comment:
True, though need is still technically defined. The variable defined by the for loop is still accessible after the for loop exits.

Try the following in your python shell to see:
```python
for a in range(0,2):
  print(a)
print(a)
```
Output will be:
```
0
1
1
```